test/modules/generic-mqtt-tester/src/message_channel.rs

```diff
@@ -58,26 +51,22 @@ impl MessageHandler for ReportResultMessageHandler {
         let sequence_number = parse_sequence_number(&received_publication);
         let batch_id = Uuid::from_u128_le(received_publication.payload.slice(4..20).get_u128_le());
 
-        if self.batch_id == batch_id {
```

Comment:
Removing this batch id check because the new Receive mode cannot do this (it is not generating the original telemetry and therefore doesn't know the batch id).